### PR TITLE
feat: users-ms-flask REST

### PR DESCRIPTION
**Pull request by zhiweit**
## Context
REST for users microservice.

## REST endpoints to test
Refer to `README.md` for the setup instructions.
- [x] `GET /users` to get all users
- [x] `GET /users/:id` to get user by id
- [x] `POST /users` to create user
- [x] `GET /users/verifiers` to get all verifiers
- [x] `POST /users/:id/offset` to create a offset for a user (after payment success)
- [x] `GET /users?milestone_id=...` to get all buyers by milestone id
- [x] `PATCH /users/:id/offset/:payment_id` to update hte buyer offset status to `confirmed` (if milestone is met) or `refund` (if milestone is rejected)

I have updated the response body on Users MS notion

### CODE REVIEW

File: users-flask/app.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+@app.route("/users/verifiers")
+def find_verifiers():
+    verifiers = db.session.execute(db.select(User).where(User.is_verifier)).scalars().all()
+    data = [v.json() for v in verifiers]
+    res = ResponseBodyJSON(True, data).json()
+    return jsonify(res), 200
+
+
+@app.route("/users/<uuid:id>/offset", methods=['POST'])
```
Comment (engzongwei) [inline users-flask/app.py:99]: Postman test data for POST: offset - create one should be updated to the following
`
{
    "amount": 10.2,
    "buyer_id": "da686e54-83d2-471b-90f6-404e2077ac59",
    "created_at": "2023-03-31T18:48:39Z",
    "milestone_id": "ed9e5902-351d-4c72-9129-f98f5dcc1f7d",
    "payment_id": "pi_3MrCv6GehkGRMk671H2IKsDX",
    "status": "pending",
    "updated_at": "2023-03-31T18:48:39Z"
}
`

File: users-flask/app.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    
+    buyer: User or None = db.session.execute(db.select(User).where(User.id == id)).scalars().first()
+    if buyer is None:
+        abort(404, description=f"User {str(id)} not found.")
+        
+    body = request.get_json()
+    
+    status = body['status']
+    if status not in OffsetStatus.values():
+        abort(400, description=f"Invalid status '{status}'. Status can only be the following: {OffsetStatus.values()}")
```
Comment (engzongwei) [inline users-flask/app.py:133]: Test data for PATCH: update buyer offset change to:
`
{
    "status": 'pending' | 'confirmed' | 'refund'

`

**Review by engzongwei — COMMENTED** [review]
I have tested the PR with postman and all endpoints looks good. Just have to change the testing data for postman and all is good

File: users-flask/config/rabbitmq_setup.py
[the remark this answers is not in the document]
Comment (zhiweit) [reply users-flask/config/rabbitmq_setup.py:1]: Currently no

**Review by ZheTaoGuo — APPROVED** [review]

**Pull request merged**